scripts/find_unique_isoform.py

```diff
@@ -90,34 +97,71 @@ def readSeqsFunc(inBed, manifest):
         # config sequences
         for locus in outDict.keys():
 
-            outDict[locus][sample] = {}
-
+            checkList = []
             for read in samfile.fetch(locus[0], int(locus[1]), int(locus[2])):
-                qname = read.query_name.split('_')[2]
-                outDict[locus][sample][qname] = {}
-                outDict[locus][sample][qname]['seq'] = seqDict[qname]
-                outDict[locus][sample][qname]['start'] = read.reference_start
-                outDict[locus][sample][qname]['query_length'] = read.query_length
+
+                #flag = '0'*(12-len(format(read.flag, 'b'))) + format(read.flag, 'b')
+                #if read.mapping_quality < 60: continue # filter by mapping quality 60
+                #if flag[0] == 1: continue # ignore supplementary alignment
+                #if flag[3] == 1: continue # ignore non-primary alignment
+
+                temp = isoform()
+                temp.sample = sample
+                temp.name = read.query_name.split('_')[2]
+                temp.id = read.query_name
+                if temp.id in checkList: continue
+                checkList.append(temp.id)
+                temp.seq = seqDict[temp.name]
+                temp.contig = locus[0]
+                temp.start = read.reference_start + 1 # convert 0-based to 1-based
+                temp.query_length = read.query_length
+                outDict[locus].append(temp)
 
         samfile.close()
 
     return outDict
 
 
 # function for unique isoform search and N-W ed stats
-def compareFunc(seqDict, outFile, minPercent):
+def compareFunc(seqDict, outPre, minPercent):
 
     '''
     find unique isoforms and corresponding N-W alignment stats.
```

Comment:
Are you using vscode? I'm not familiar with this docstring formatting. But, we should just choose a style.

this plugin generates docstrings for you from the function definition

https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring

you can configure a number of different styles in the vscode settings.json.

[Default style looks like this](https://github.com/collaborativebioinformatics/isocomp/blob/ea01bdd9579330ca79ae521f8e8b66ecb41dd249/src/isocomp/create_windows.py#L29)